### ISSUE TITLE
Tracing programs must provide btf_id

### ISSUE THREAD

**Issue by ZhangYet** [bpftrace/bpftrace#3386]
When running the bpftrace script below:

```
build/src/bpftrace -e 'kfunc:filp_close { print(path(args->filp->f_path)); }'
```
I encounter error when loading. The output with `-v` is as below:
```
Kernel error log:
Tracing programs must provide btf_id
processed 0 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0

```

After `git bisect`, I found that commit 66eac00c imported this problem.

### What reproduces the bug? Provide code if possible.

```shell
git checkout 66eac00c
cmake -B ./build -DBUILD_TESTING=OFF -DCMAKE_EXPORT_COMPILE_COMMANDS=1
make -C ./build -j$(nproc)

build/src/bpftrace -e 'kfunc:filp_close { print(path(args->filp->f_path)); }'
```

### `bpftrace --info` output
```
System
  OS: Linux 6.10.2-arch1-1 #1 SMP PREEMPT_DYNAMIC Sat, 27 Jul 2024 16:49:55 +0000
  Arch: x86_64

Build
  version: v0.21.0-28-g66ea
  LLVM: 18.1.8
  unsafe probe: no
  bfd: yes
  liblldb (DWARF support): no
  libsystemd (systemd notify support): no

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes
  get_boot_ns: yes
  dpath: yes
  skboutput: yes
  get_tai_ns: yes
  get_func_ip: yes
  jiffies64: yes
  for_each_map_elem: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: yes
  module btf: yes
  map batch: yes
  uprobe refcount (depends on Build:bcc bpf_attach_uprobe refcount): yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes
  ringbuf: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes
  kprobe_multi: yes
  uprobe_multi: yes
  raw_tp_special: yes
  iter: yes

```

**Comment by jordalgo** [comment]
@ZhangYet Looks like the same issue as here: https://github.com/bpftrace/bpftrace/issues/3317
TL;DR - you need to update your version of libbpf

**Comment by ZhangYet** [comment]
@jordalgo That's it! Thanks.
